src/check/property/Property.generic.ts

```diff
@@ -59,16 +60,23 @@ export interface IPropertyWithHooks<Ts> extends IProperty<Ts> {
  * @internal
  */
 export class Property<Ts> implements IProperty<Ts>, IPropertyWithHooks<Ts> {
-  static dummyHook: PropertyHookFunction = () => {};
-  private beforeEachHook: PropertyHookFunction = Property.dummyHook;
-  private afterEachHook: PropertyHookFunction = Property.dummyHook;
-  constructor(readonly arb: Arbitrary<Ts>, readonly predicate: (t: Ts) => boolean | void) {}
+  static dummyGlobalHook: GlobalPropertyHookFunction = () => {};
+  static defaultHook: PropertyHookFunction = (globalHook) => globalHook();
+  private readonly globalBeforeEachHook: GlobalPropertyHookFunction;
+  private readonly globalAfterEachHook: GlobalPropertyHookFunction;
+  private beforeEachHook: PropertyHookFunction = Property.defaultHook;
+  private afterEachHook: PropertyHookFunction = Property.defaultHook;
+  constructor(readonly arb: Arbitrary<Ts>, readonly predicate: (t: Ts) => boolean | void) {
+    const { beforeEach = Property.dummyGlobalHook, afterEach = Property.dummyGlobalHook } = readConfigureGlobal() || {};
+    this.globalBeforeEachHook = beforeEach;
+    this.globalAfterEachHook = afterEach;
```

Comment:
As for the async version:
```suggestion
    const { asyncBeforeEach,  asyncAfterEach,  beforeEach,  afterEach  } = readConfigureGlobal() || {};
    if (asyncBeforeEach !== undefined) {
      // Why this check?
      // - Users might have been told that a clean-up hook was globally set. So write her/his property
      //   thinking that the hook would be there. But the hook was actually defined on async versions only so it passes
      // - Consistency with the rest of the philosophy of the lib and where it goes concerning sync vs async
      // - As a library authors, it's easier to relax constraints that add new ones (breaking change). So putting it now
      // would at least require a change given there are good reasons not to do that. Not putting it would be impossible
      // to change
      throw new Error('Illegal');
    }
    if (asyncAfterEach !== undefined) {
      throw new Error('Illegal');
    }
    this.beforeEachHook = beforeEach || Property.defaultHook;
    this.afterEachHook = afterEach || Property.defaultHook;
```

Reply:
Fair enough! I'll change it